packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -979,7 +1097,6 @@ class _DropdownButtonFormFieldState<T> extends FormFieldState<T> {
   @override
   void didChange(T value) {
     super.didChange(value);
-    if (widget.onChanged != null)
-      widget.onChanged(value);
+    widget.onChanged(value);
```

Comment:
Since onChanged is handled like this `onChanged: onChanged==null? null : field.didChange(value)`. I guess `onChanged` won't be null here.

Reply:
As far as I understand, this makes sense because `field.didChange` is only ever called when `onChanged` is not `null`... although reading the logic at first glance was somewhat confusing.

Reply:
Done